stateless.go

```diff
@@ -813,10 +806,7 @@ func (n *StatelessNode) toInternalNode() *InternalNode {
 		if child, ok := n.children[byte(i)]; ok {
 			internal.children[i] = child
 		} else if serialized, ok := n.unresolved[byte(i)]; ok {
-			hashed := &HashedNode{hash: new(Fr), commitment: new(Point)}
-			hashed.commitment.SetBytesTrusted(serialized)
-			toFr(hashed.hash, hashed.commitment)
-			internal.children[byte(i)] = hashed
+			internal.children[byte(i)] = &HashedNode{serialized}
```

Comment:
this whole function can be deleted because it was only used in one test, that has been modified